src/OAuth2/ResponseType/AccessToken.php

```diff
@@ -153,4 +153,37 @@ protected function generateRefreshToken()
     {
         return $this->generateAccessToken(); // let's reuse the same scheme for token generation
     }
+
+    /**
+     * Handle the revoking of refresh tokens, and access tokens if supported / desirable
+     * RFC7009 specifies that "If the server is unable to locate the token using
+     * the given hint, it MUST extend its search across all of its supported token types"
+     *
+     * @param $token
+     * @param $tokenTypeHint
+     * @param null $skip
+     * @return mixed
+     */
+    public function revokeToken($token, $tokenTypeHint, $skip = null)
+    {
+        $revoked = false;
+
+        if ($skip !== 'refresh_token' && (($tokenTypeHint === 'refresh_token' && $this->refreshStorage) || $tokenTypeHint === null)) {
+            $revoked = $this->refreshStorage->unsetRefreshToken($token);
+        } elseif ($skip !== 'access_token' && ($tokenTypeHint === 'access_token' || $tokenTypeHint === null)) {
+            // @todo remove this check for v2.0
+            if (!method_exists($this->tokenStorage, 'unsetAccessToken')) {
+                $class = get_class($this->tokenStorage);
+                throw new \RuntimeException("Token storage {$class} does not implement required unsetAccessToken method");
+            }
+            $revoked = $this->tokenStorage->unsetAccessToken($token);
+        }
+
+        if (!$revoked && $tokenTypeHint !== null) {
+            // expand search to all token types, skipping the one we've already tried
+            return $this->revokeToken($token, null, $tokenTypeHint);
+        }
+
+        return $revoked;
+    }
```

Comment:
Hi, I'm merging your PR into my forked oauth repo as I needed it asap, and I was just writing some extra test and noticed some potential issue.

In this function, if you would pass some invalid ```$tokenTypeHint``` value in the first call (something different from "refresh_token" or "access_token") and the token you want to revoke would be stored in the tokenStorage, the expanded search for all token types would not work, as it would just retry once more always trying the refreshStorage and not finding anything.
If the token would be stored in refreshStorage, then it would succeed deleting it, out of luck.

I noticed that AccessTokenTest is not really testing AccessToken class, but instead just the storage underneath it. So I noticed this when writing some test for the AccessToken class itself, which would cover the behaviour of revokeToken function.